nexus/src/app/sled.rs

```diff
@@ -208,8 +209,10 @@ impl super::Nexus {
         opctx: &'a OpContext,
         disk_selector: &params::PhysicalDiskPath,
     ) -> Result<lookup::PhysicalDisk<'a>, Error> {
-        Ok(lookup::LookupPath::new(&opctx, &self.db_datastore)
-            .physical_disk(disk_selector.disk_id))
+        // XXX how to do typed UUID as part of dropshot path?
```

Comment:
This is one thing I didn't know how to do, or if dropshot even supported it. Comments welcome :)

Reply:
It should just work if you change the type of `params::PhysicalDiskPath::disk_id` to be `PhysicalDiskUuid` instead of `Uuid`. However, I'm not sure we want to: this would affect the public API and would "leak" our typed UUIDs:

```diff
--- a/openapi/nexus.json
+++ b/openapi/nexus.json
@@ -4786,8 +4786,7 @@
             "description": "ID of the physical disk",
             "required": true,
             "schema": {
-              "type": "string",
-              "format": "uuid"
+              "$ref": "#/components/schemas/TypedUuidForPhysicalDiskKind"
             }
           }
         ],
@@ -22499,6 +22498,10 @@
           "ram_provisioned"
         ]
       },
+      "TypedUuidForPhysicalDiskKind": {
+        "type": "string",
+        "format": "uuid"
+      },
       "NameSortMode": {
         "description": "Supported set of sort modes for scanning by name only\n\nCurrently, we only support scanning in ascending order.",
         "oneOf": [
```

I vaguely remember @ahl suggesting this wasn't something we wanted, but can't find the details of that suggestion at the moment. Maybe this doesn't translate particularly well for non-Rust OpenAPI clients?